<launch>
    <arg name="model" default="2024_gate_glyphs" description='{"options":["yolov7_tiny_2023_main","yolov8n_2023_main","gate_buoy","gate"],"allowEmpty":false,"help":"The DepthAI model to use from depthai_models.yaml"}'/>
    <arg name="rgb_raw" default="True" description='{"type":"bool","allowEmpty":false,"help":"Whether to publish the raw rgb stream (no detections)."}'/>
    <arg name="rgb_detections" default="True" description='{"type":"bool","allowEmpty":false,"help":"Whether to publish the rgb stream with detections overlaid."}'/>
    <arg name="depth" default="False" description='{"type":"bool","allowEmpty":false,"help":"Whether to publish the depth map."}'/>
    <arg name="sync_nn" default="True" description='{"type":"bool","allowEmpty":false,"help":"Whether to sync the RGB frames with the detections so the latest published detections correspond to the latest published RGB frame."}'/>
    <arg name="using_sonar" default="False" description='{"type":"bool","allowEmpty":false,"help":"Whether to use sonar to estimate depth instead of stereo."}'/>
    <arg name="show_class_name" default="True" description='{"type":"bool","allowEmpty":false,"help":"Whether to show the class name."}'/>
    <arg name="show_confidence" default="True" description='{"type":"bool","allowEmpty":false,"help":"Whether to show the confidence value."}'/>
    <arg name="correct_color" default="True" description='{"type":"bool","allowEmpty":false,"help":"Whether to apply underwater color correction."}'/>
    <node name="depthai_spatial_detection" exec="depthai_spatial_detection" pkg="cv" output="screen">
        <param name="model" type="string" value="$(var model)" />
        <param name="rgb_raw" type="bool" value="$(var rgb_raw)" />
        <param name="rgb_detections" type="bool" value="$(var rgb_detections)" />
        <param name="depth" type="bool" value="$(var depth)" />
        <param name="sync_nn" type="bool" value="$(var sync_nn)" />
        <param name="using_sonar" type="bool" value="$(var using_sonar)" />
        <param name="show_class_name" type="bool" value="$(var show_class_name)" />
        <param name="show_confidence" type="bool" value="$(var show_confidence)" />
        <param name="correct_color" type="bool" value="$(var correct_color)" />
    </node>
</launch>
